core/src/main/java/org/eclipse/dash/licenses/http/HttpClientService.java

```diff
@@ -67,7 +68,16 @@ public int post(String url, String contentType, String payload, Consumer<String>
 					Thread.sleep(1000 * tries);
 					continue;
 				}
-				
+				if (response.statusCode() == 429 && tries++ < maxRatelimitRequestsPerExecure) {
+					long saftyMargin = 100l;
+					logger.info("HTTP response 429 (ratelimit)");
+					Optional<String> oResetTime = response.headers().firstValue("x-ratelimit-reset");
+					Long resetTime = oResetTime.map(Long::valueOf).orElseGet(System::currentTimeMillis);
+					Long sleepTime = resetTime - System.currentTimeMillis() + saftyMargin;
+					logger.info("x-ratelimit-reset needs a sleep for: " + sleepTime + " ms");
+					Thread.sleep(sleepTime);
```

Comment:
I'm already uncomfortable with possibly waiting 10 seconds to bail after repeated 502s. This could -- under a very extreme set of circumstances -- block for 5 minutes. It's unlikely, but it is possible.

IMHO, we need to be able to let the user decide how to handle this. We could, for example, add a `-D` parameter to configure how many times to retry. With this first implementation, the default should be `0`.